src/pages/home/ReportScreen.js

```diff
@@ -160,6 +160,14 @@ function ReportScreen({
     const [listHeight, setListHeight] = useState(0);
 
     const reportID = getReportID(route);
+
+    useEffect(() => {
+        if (!reportID) {
+            return;
+        }
+        Report.updateLastVisitTime(reportID);
+    }, [reportID]);
+
```

Comment:
@techievivek updated to just update the ```lastVisitTime``` if the reportID changes

Reply:
Is there any specific reason to not put lastVisitTime in OpenReport (maybe in optimistic onyx update)? I am thinking only offline behaviour needs to be checked in this.

Reply:
@shubham1206agra There are a few cases we do not call the ```OpenReport``` API

Reply:
Can you point out the cases?

Reply:
https://github.com/Expensify/App/blob/f52c59d55be2d6d2775623d92c3043bd6d42cc0f/src/pages/home/report/ReportActionsView.js#L105-L112

Reply:
Pls help check the above case

Reply:
But in this, don't we have `lastReadTime` as the latest?

Reply:
Before, I tried the solution you`ve mentioned " put lastVisitTime in OpenReport", and when testing, I encountered a few cases that the OpenReport is not called. So we need to make sure the ```lastVisitTime``` is always updated once we open report

Reply:
Can you tell me more about these cases?

Reply:
For example, we create new report in offline mode

Reply:
Is there any reason to not use the stack navigator listener?

Reply:
In stack navigation, the ```Report.updateLastVisitTime(reportID);``` is called once the navigation stack changes, even if the user just navigates to other screens like setting, profile, ...

Reply:
> For example, we create new group in offline mode, so the field `props.report.isOptimisticReport` is true

But here we should have `lastReadTime` here

Reply:
@shubham1206agra What do you mean?

Reply:
https://github.com/Expensify/App/pull/31424#discussion_r1403927078

Reply:
@shubham1206agra There is the case:
1. Force offline mode
2. Create group A (report A)
3. Open report B and send message
4. Open report A again (The ```lastVisitTime``` is not updated if follow this [solution](https://github.com/Expensify/App/pull/31424#discussion_r1403923833)

Reply:
@DylanDylann, I am actually not a fan of the current solution because if we try to open an invalid report, then it should not open any report, but it will create a report under `report_` collection, which is undesired in any case, and may cause errors.

Reply:
the use effect maybe not triggered when pressing browser back button as the previous ReportScreen is cached and not updated

Reply:
@shubham1206agra Updated the PR to fix your comments:
1. Does not update ```lastVisitTime``` with the empty report
2. Update the ```lastVisitTime``` when going back

Reply:
Please help check

Reply:
@shubham1206agra Please help review once you have a chance

Reply:
I recommend utilizing the Navigation state event and filtering out states other than Report.

Reply:
@getusha It leads to update ```lastVisitTime``` with the empty report

Reply:
@DylanDylann why can't we just filter empty reports? or am i missing something?

Reply:
@getusha Maybe cannot. Currently, when we open the invalid report, it still call API to get this report. Then, if the returned report does not have ```report.reportID```, then we know it is an invalid report and display "Not found view"